File: 2023-12-21/leetcode_938.js
const rangeSumBST = (root, low, high) => {
  let sum = 0;

  const dfs = (node) => {
    if (!node) {
      return 0;
    }
    if (!!node.val && node.val >= low && node.val <= high) {
      sum += node.val;
    }
    dfs(node.left);
    dfs(node.right);
  };
  dfs(root);
  return sum;
};

File: leetcode/leetcode_938.py
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        self.sum = 0
        def dfs(node):
            if not node:
                return
            if node.val and node.val >= low and node.val <= high:
                self.sum += node.val
            dfs(node.left)
            dfs(node.right)
        dfs(root)
        return self.sum